File: Apps/Codaxy.Dextop.Showcase/Demos/Remoting/HelloRemotingDemo.js
Ext.define('Showcase.demos.HelloRemotingWindow', {
    extend: 'Dextop.Window',
    width: 200,
    height: 150,
    title: 'Hello Remoting Demo',
    
    initComponent: function () {
        Ext.apply(this, {
            bodyStyle: 'padding: 5px',
            layout: {
				type: 'vbox',
				align: 'stretch'
			},
			defaults: {
				xtype: 'button',
				margins: '3 3 3 3',
				flex: 1,
				scope: this,
			},
            items: [{                
                text: 'Say Hello',                
                handler: function () {
                    this.remote.GetHelloWorldMessage(function (r) {
                        if (r && r.success)
                            alert(r.result);
                        else
                            alert('failed');
                    });
                }
            },{                
                text: 'Search for John Smith',                
                handler: function () {
                    this.remote.Search({FirstName: 'John',LastName: 'Smith'}, function (r) {
                        if (r && r.success)
                            alert(r.result);
                        else
                            alert('failed');
                    });
                }
            }, {                
				text: 'Request Server Notification',                
                handler: function () {
                    this.remote.RequestServerNotification(function (r) {
                        if (!r || !r.success)
                            alert('failed');
                    });
                }
            }]
        });
        this.callParent(arguments);
    },

    onServerMessage: function(m) {
    	if (m.type === 'alert')
    		alert(m.msg);	
    }
});

File: Apps/Codaxy.Dextop.Showcase/Demos/Remoting/HelloRemotingDemo.cs
using System;
using System.Threading;
using Codaxy.Dextop.Remoting;

namespace Codaxy.Dextop.Showcase.Demos.Remoting
{
    [Demo("HelloRemotingDemo",
        Title = "Hello Remoting",
        Description = "Making remote method calls is really easy with Dextop.",
        Path = "~/Demos/Remoting"
    )]
    [LevelBeginner]
    [TopicDextopRemoting]
	[CategoryGettingStarted]
    public class HelloRemotingWindow : DextopWindow
    {
        public override void InitRemotable(DextopRemote remote, DextopConfig config)
        {
            base.InitRemotable(remote, config);
            Remote.RemoteHostType = "Showcase.demos.HelloRemotingWindow";            
        }

        [DextopRemotable]
        public String GetHelloWorldMessage() { return "Hello World!"; }

        class SearchFilter
        {
            public String FirstName { get; set; }
            public String LastName { get; set; }
        }

        [DextopRemotable]
        String Search(SearchFilter data)
        {
            return String.Format("Information about '{0} {1}' not found in the database.", data.FirstName, data.LastName);
        }

        [DextopRemotable]
        void RequestServerNotification()
        {
            ThreadPool.QueueUserWorkItem((state) =>
            {
                try
                {
                    Thread.Sleep(3000);
                    var msg = new { type = "alert", msg = "This is a notification you requested 3 seconds ago." };
                    Remote.SendMessage(msg);
                }
                catch { }
            });
        }
    }
}